CMake/Findpyarrow.cmake

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+find_package(Python REQUIRED COMPONENTS Interpreter)
+
+execute_process(
+  COMMAND
+    "${Python_EXECUTABLE}" -c "\
+import pyarrow
+print(pyarrow.get_include())
+"
+  OUTPUT_VARIABLE _pyarrow_include_dir
+  OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+execute_process(
+  COMMAND
+    "${Python_EXECUTABLE}" -c "\
+import pyarrow
```

Comment:
Should we do some version check for pyarrow ? Also what happens if you change venv's etc . Curious what other native python packages which have to depend on pyarrow.

Reply:
If there is a version bound we can add it to our pyproject.toml, no problem. 

Well pyarrow is a dependency so will always be installed when building the package/installing the package through python means, this will just give us the paths to link against while building. When installing the package pyarrow will also be installed and we then jsut find it through rpath.

Pretty sure they do this, after all that's why pyarrow added these functions ^^ though I think it might make sense to have them generate a cmake config so we can drop the `Findpyarrow.cmake` :D